src/utils/timeout.ts

```diff
@@ -16,9 +16,13 @@ limitations under the License.
 */
 
 import {ConnectionError} from "../matrix/error.js";
+import type {Timeout} from "../platform/web/dom/Clock.js"
 
+type TimeoutCreator = (ms: number) => Timeout;
+// ts-todo: export type RequestResult from fetch.js? we'll need to wait until it's typescript though.
+type Abortable = { abort(): void; [key: string]: any };
```

Comment:
this type already exists in `AbortableOperation`, you could export it there and use it here.